### PR TITLE
Introduce Github Actions to CryptoAnalysis

### PR DESCRIPTION
**Pull request by svenfeld**
This PR adds github actions to CryptoAnalysis. With this PR CryptoAnalysis is build and tested by the CI of Github Actions.

### CODE REVIEW

File: .github/workflows/main_build.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: CryptoAnalysis build
+
+on: [push, pull_request]
```
Comment (AnakinSklavenwalker) [inline .github/workflows/main_build.yml:3]: maybe restrict this to develop and main only. Feature braches don't need to waste runner performence.

File: .github/workflows/main_build.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v2
+      # Sets up Java version
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-package: jdk
+          java-version: '8'
```
Comment (AnakinSklavenwalker) [inline .github/workflows/main_build.yml:22]: can we use java 11?

File: .github/workflows/main_build.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      # Restores Maven dependecies
+      - name: Restore local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Run maven command
+        run: mvn clean install
```
Comment (AnakinSklavenwalker) [inline .github/workflows/main_build.yml:37]: would `mvn clean verify` be sufficent already, or is this for later deploy use already?

**Review by AnakinSklavenwalker — CHANGES_REQUESTED** [review]
General idea, what do you think about splitting the run into a full test job (like this)
and a separate job that just runs `mvn clean compile`. This way we have a fast failure cycle if things break. The compile job should be runned before or in parallel to this job. 

Also, can you revmove jekins related stuff here already?

File: .github/workflows/main_build.yml (hunk 1 above)
Comment (svenfeld) [reply .github/workflows/main_build.yml:3]: I added it like this so that forks also run the tests on their branches.

**Comment by svenfeld** [comment]
I don't think splitting in seperate jobs is beneficial for a faster failure cycle since for the tests to run the compilation has to be successful. A failure in that phase would result in an aborted run anyway. However, I agree that using `mvn clean verify` or `mvn clean test` is the better choice than `mvn clean install`

File: .github/workflows/main_build.yml (hunk 1 above)
Comment (schlichtig) [reply .github/workflows/main_build.yml:3]: We can try it like that first. If we run into issues, we will restrict this.

**Review by schlichtig — APPROVED** [review]

**Pull request merged**